### PR TITLE
added openocean dex

### PR DESCRIPTION
**Pull request by ARDev097**
added trades data for openocean dex

### CODE REVIEW

**Comment by ARDev097** [comment]
Hello @Hosuke ,
I have added spellbook for openocean dex, but all checks are not running due to some conflicts. Please look into it.

File: models/openocean/optimism/openocean_v2_optimism_trades.sql [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    ON erc20b.contract_address = dexs.token_sold_address
+    AND erc20b.blockchain = 'optimism'
+LEFT JOIN {{ source('prices', 'usd') }} p_bought
+    ON p_bought.minute = date_trunc('minute', dexs.block_time)
+    AND p_bought.contract_address = dexs.token_bought_address
+    AND p_bought.blockchain = 'optimism'
+    {% if not is_incremental() %}
+    AND p_bought.minute >= timestamp '{{project_start_date}}'
+    {% else %}
+    AND p_bought.minute >= date_trunc('day', now() - interval '7' day)
```
Comment (Hosuke) [inline models/openocean/optimism/openocean_v2_optimism_trades.sql:90]: Here as well.

File: models/openocean/optimism/openocean_v2_optimism_trades.sql [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    {% endif %}
+LEFT JOIN {{ source('prices', 'usd') }} p_sold
+    ON p_sold.minute = date_trunc('minute', dexs.block_time)
+    AND p_sold.contract_address = dexs.token_sold_address
+    AND p_sold.blockchain = 'optimism'
+    {% if not is_incremental() %}
+    AND p_sold.minute >= timestamp '{{project_start_date}}'
+    {% else %}
+    AND p_sold.minute >= date_trunc('day', now() - interval '7' day)
```
Comment (Hosuke) [inline models/openocean/optimism/openocean_v2_optimism_trades.sql:99]: Same as above.

**Review by Hosuke — CHANGES_REQUESTED** [review]

File: models/openocean/optimism/openocean_optimism_schema.yml [hunk 3]
```diff
@@ -0,0 +1,100 @@
+version: 2
+
+models:
+  - name: openocean_v2_optimism_trades
```
Comment (Hosuke) [inline models/openocean/optimism/openocean_optimism_schema.yml:4]: We can add roll-up spells like `openocean.trades` or `openocean_optimism.trades`, if there will be other versions of the dex.

**Review by Hosuke — CHANGES_REQUESTED** [review]

File: models/openocean/optimism/openocean_optimism_trades.sql [hunk 4]
```diff
@@ -0,0 +1,100 @@
+{{ config(
+    tags = ['dunesql'],
+    schema = 'openocean_v2_optimism',
```
Comment (Hosuke) [inline models/openocean/optimism/openocean_optimism_trades.sql:3]:
```suggestion
    schema = 'openocean_optimism',
```

**Review by Hosuke — APPROVED** [review]
LGTM.✅
Thank you @ARDev097 

[Sample openocean_optimism_trades](https://dune.com/queries/3132468)

File: models/openocean/optimism/openocean_optimism_trades.sql [hunk 5]
```diff
@@ -0,0 +1,98 @@
+{{ config(
+    tags = ['dunesql'],
+    schema = 'openocean_optimism',
+    alias = alias('trades'),
+    partition_by = ['block_month'],
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['block_date', 'blockchain', 'project', 'version', 'tx_hash', 'evt_index', 'trace_address']
+    )
```
Comment (Hosuke) [inline models/openocean/optimism/openocean_optimism_trades.sql:10]: By the way we may add a post_hook here to expose this model.
And `trace_address` may not be needed anymore.
```suggestion
    unique_key = ['block_date', 'blockchain', 'project', 'version', 'tx_hash', 'evt_index'],
    post_hook='{{ expose_spells(\'["optimism"]\',
                                "project",
                                "openocean",
                                \'["ARDev097"]\') }}'
    )
```

File: models/openocean/optimism/openocean_optimism_schema.yml [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+    tests:
+      - dbt_utils.unique_combination_of_columns:
+          combination_of_columns:
+            - block_date
+            - blockchain
+            - project
+            - version
+            - tx_hash
+            - evt_index
+            - trace_address
```
Comment (Hosuke) [inline models/openocean/optimism/openocean_optimism_schema.yml:23]:
```suggestion
```

**Review by jeff-dude — APPROVED** [review]
we can remove `trace_address` when we refactor for new design in the near future, otherwise lgtm and we can merge 👍

**Pull request merged**